connectors/microsoft/azure-blobstorage/src/main/java/io/camunda/connector/azure/blobstorage/model/response/Element.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. Licensed under a proprietary license.
+ * See the License.txt file for more information. You may not use this file
+ * except in compliance with the proprietary license.
+ */
+package io.camunda.connector.azure.blobstorage.model.response;
+
+import io.camunda.document.Document;
+
+public interface Element {
+  record DocumentContent(Document document) implements Element {}
+
+  record StringContent(String content) implements Element {}
+
+  record JsonContent(Object content) implements Element {}
+}
```

Comment:
What about a sealed class?  Then you can also skip the `element` property and have the implementations on the top level. See https://github.com/camunda/connectors/blob/main/connectors/box/src/main/java/io/camunda/connector/box/model/BoxResult.java for an example.